src/main/java/de/tum/in/www1/artemis/service/plagiarism/ContinuousPlagiarismControlResultsService.java

```diff
@@ -0,0 +1,102 @@
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+import de.tum.in.www1.artemis.domain.enumeration.Visibility;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismResult;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+import de.tum.in.www1.artemis.repository.SubmissionRepository;
+
+/**
+ * Manages results of checks executed during continuous plagiarism control.
+ */
+@Service
+@Component
+class ContinuousPlagiarismControlResultsService {
+
+    private static final Logger log = LoggerFactory.getLogger(ContinuousPlagiarismControlResultsService.class);
+
+    private static final String PLAGIARISM_RESULT_FEEDBACK_EN = "Suspicion of plagiarism! Score reduced to 0.";
+
+    private static final String PLAGIARISM_RESULT_FEEDBACK_DE = "Verdacht auf Plagiat! Punktestand auf 0 reduziert.";
+
+    private final SubmissionRepository submissionRepository;
+
+    private final ResultRepository resultRepository;
+
+    ContinuousPlagiarismControlResultsService(SubmissionRepository submissionRepository, ResultRepository resultRepository) {
+        this.submissionRepository = submissionRepository;
+        this.resultRepository = resultRepository;
+    }
+
+    void handleCpcResult(PlagiarismResult<?> result) {
+        var submissionsIdsWithPlagiarism = result.getComparisons().stream().flatMap(it -> Stream.of(it.getSubmissionA().getSubmissionId(), it.getSubmissionB().getSubmissionId()))
+                .collect(toUnmodifiableSet());
+
+        result.getExercise().getStudentParticipations().stream().filter(participation -> participation.findLatestSubmission().isPresent()).forEach(participation -> {
+            var submission = participation.findLatestSubmission().get();
+
+            boolean plagiarismDetected = submissionsIdsWithPlagiarism.contains(submission.getId());
+            submissionRepository.setPlagiarismDetected(plagiarismDetected, submission.getId());
+
+            if (plagiarismDetected) {
+                addResultWithPlagiarismFeedback(submission, participation);
+            }
+            else {
+                deletePastResultsWithPlagiarismFeedback(submission.getId());
+            }
+        });
+    }
+
+    private void addResultWithPlagiarismFeedback(Submission submission, StudentParticipation participation) {
+        log.debug("Adding cpc results for submission: submissionId={}.", submission.getId());
+
+        var result = new Result();
+        result.setSubmission(submission);
+        result.setAssessmentType(AssessmentType.AUTOMATIC);
+        result.setCompletionDate(ZonedDateTime.now());
+        result.setParticipation(participation);
+        result.setRated(true);
+        result.setExampleResult(false);
+        result.setSuccessful(false);
+        result.setScore(0.0, participation.getExercise().getCourseViaExerciseGroupOrCourseMember());
+
+        var feedback = new Feedback();
```

Comment:
Add an issue for adding a constructor for Feedback in order to avoid calling all the upcomming setters.

Reply:
https://github.com/ls1intum/Artemis/pull/6804#discussion_r1264676868

Reply:
#6920